### ISSUE TITLE
Provide Gradle Kotlin DSL examples in documentation

### ISSUE THREAD

**Issue by eriwen** [vanniktech/gradle-dependency-graph-generator-plugin#79]
As a [Gradle Kotlin DSL](https://github.com/gradle/kotlin-dsl) adopter, I'd like to know how to configure this plugin using the Kotlin DSL.

[This plugin was identified](https://github.com/gradle/gradle/issues/6790) as one of the most useful Gradle plugins. With Kotlin DSL 1.0 approaching shortly, it'd be wonderful if the documentation provided Kotlin and Groovy DSL samples. [Blog post with suggestions](https://blog.gradle.org/groovy-kotlin-dsl-samples).

Here are a couple additional resources:

 * [Migrating build logic from Groovy to Kotlin guide](https://guides.gradle.org/migrating-build-logic-from-groovy-to-kotlin/)
 * [Exemplar for automated samples testing](https://blog.gradle.org/documentation-samples-testing-exemplar)
 * [Join #kotlin-dsl channel in Gradle Community Slack for personalized help](https://join.slack.com/t/gradle-community/shared_invite/enQtNDE3MzAwNjkxMzY0LTYwMTk0MWUwN2FiMzIzOWM3MzBjYjMxNWYzMDE1NGIwOTJkMTQ2NDEzOGM2OWIzNmU1ZTk5MjVhYjFhMTI3MmE)

Thank you for providing a great Gradle plugin! —The Gradle Team

**Comment by vanniktech** [comment]
Thanks for reaching out. I think I'll still want to change a thing or two about the extension API and once I've done that I'll make sure to include samples for Kotlin too.

**Comment by vanniktech** [comment]
Took me a while but now with the latest SNAPSHOT this is easier since I'm using  `NamedDomainObjectContainer` - https://github.com/vanniktech/gradle-dependency-graph-generator-plugin/pull/92

The example from the README converted to Kotlin looks like this:

```kotlin
rootProject.plugins.apply(DependencyGraphGeneratorPlugin::class.java)
rootProject.configure<DependencyGraphGeneratorExtension> {
  generators.create("firebaseLibrarys") {
    include = { dependency -> dependency.moduleGroup.startsWith("com.google.firebase") }
    children = { true } // Include transitive dependencies.
    dependencyNode = { node, dependency -> node.add(Style.FILLED, Color.rgb("#ffcb2b")) } // Give them some color.
  }
}
```

**Comment by jmfayard** [comment]
@vanniktech this issue looks solved, am I right?

**Comment by vanniktech** [comment]
The new version hasn't been published

**Comment by fajarnuha** [comment]
i think we can put this into documentation already @vanniktech

**Comment by St4B** [comment]
@vanniktech Are there any plans to release the new version?

**Comment by vanniktech** [comment]
I've just released 0.6.0
